posting/list.go

```diff
@@ -868,15 +868,12 @@ func (l *List) ToBackupPostingList(bl *pb.BackupPostingList, alloc *z.Allocator)
 	defer out.free()
 
 	ol := out.plist
-	// Encode uids to []byte instead of []uint64 if we have more than 1000
-	// uids. We do this to improve the memory usage.
-	if codec.ApproxLen(ol.Pack) > 1024 {
-		buf := codec.DecodeToBuffer(ol.Pack, 0)
-		defer buf.Release()
-		bl.UidBytes = buf.Bytes()
-	} else {
-		bl.Uids = codec.Decode(ol.Pack, 0)
-	}
+
+	// Encode uids to []byte instead of []uint64. This helps improve memory usage.
+	buf.Reset()
+	codec.DecodeToBuffer(buf, ol.Pack)
+	bl.UidBytes = buf.Bytes()
+
```

Comment:
Why did we change this heuristic? This change would mean that `bl.Uid` remains for backward compatibility only.